### ISSUE TITLE
Lack of documentation about: saving-restoring graphs & other languages API about this

### ISSUE THREAD

**Issue by iurilarosa** [tensorflow/tensorflow#17359]
I think that one of the main features of TensorFlow is its portability. I want to exploit it in a huge code written in C, where a small but computational heavy part can be computed using GPU through TF. So I started to study the problem about saving a graph, loading it to another code (possibly in another programming language) and passing it data to perform some computations.

I'm finding very difficult to understand the documentation. I try to point the main problems:
1- there is no C API documentation despite it is the only "other language" which satisfies the stability promises, there is only a link to the `c_api.h` file in the repo that I find very difficult to understand;
2- it is very unclear how one can save a graph which can be restored and loaded in another code written in C (or maybe C++), and I don't find where is explained how one can run the graph in the other code passing data to it. Indeed, people had to write many tutorials, or ask many questions on StackOverflow which are unfortunately very outdated (e.g. [here](https://medium.com/jim-fleming/loading-a-tensorflow-graph-with-the-c-api-4caaff88463f) and [here](https://stackoverflow.com/questions/38947658/tensorflow-saving-into-loading-a-graph-from-a-file));
3- as far as I can see there are many ways to do save/restoring but it is unclear when you should use one of these way or another. E.g. `saver.save` -> `saver.restore`; `tf.train.write_graph` -> `tf.import_graph_def` or C analogue `TF_GraphImportGraphDef`; `tf.saved_model.builder.SavedModelBuilder` -> `saved_model.loader.load` or C analogue `LoadSavedModel`.

The more I explore the documentation about this topic, the more questions arises, but I hope to have explained enough the issues I'm facing studying it. Shortly, I could say simply that we need a schematic and coherent page which can summarize the procedures about saving, loading, running graph at least from-to any languages which satisfy the stability promises, maybe with links to the specific standard methods and explanation about which of these ones could be used in a certain situation and why.

**Comment by PatWie** [comment]
The C_API is ready to understand. It is an issue of writing in C such that the LOC is very large. But it is pretty straightforward to restore a graph in different languages.

[These examples for C,C++,Go](https://github.com/PatWie/tensorflow_inference) work. Instead of wasting time by reading "tutorials", reading the TF source is more helpful. You probably cannot write a a small guide for loading a graph in C.

There is only one way of restoring a graph. The other stuff is syntactic sugar. In your issue description your probably confuse restoring a graph vs. restoring parameters.

**Comment by iurilarosa** [comment]
Indeed I can surely admit that I'm not so good understanding codes "from scratch" and I can be quite confused, but since i think in general this could be true for many people, this is the reason why well written documentations are so important. I'd like to contribute but first I need to understand the subject :)

Apart the general issue, I thank you for your answer, I'm watching your examples and they are interesting.

**Comment by PinkySan** [comment]
Hey @iurilarosa ,

i am having the same issues as you. I do not really like the documentation for C++. Therefore i created a litte github repository for dealing tensorflow with C++. Unfortunatly i am stucked with the restoring problem. At least i was able to save some checkpoints. (But i am using CMake)

https://github.com/PinkySan/TensorflowHandlingTests

If you want to share it and support.

**Comment by tatianashp** [comment]
@sukritiramesh Can you help?

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatianashp: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatianashp: It has been 15 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatianashp: It has been 36 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by sonu107** [comment]
[To restore the model and use it this is a good blog.](http://cv-tricks.com/tensorflow-tutorial/save-restore-tensorflow-models-quick-complete-tutorial/)

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatianashp: It has been 15 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatianashp: It has been 30 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatianashp: It has been 44 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatianashp: It has been 60 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatianashp: It has been 75 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatianashp: It has been 90 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatianashp: It has been 105 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatianashp: It has been 119 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.
